src/helpers/decorators.ts

```diff
@@ -0,0 +1,21 @@
+import { type Purchases } from "../main";
+
+export function requiresInitialization<
+  This extends Purchases,
+  Args extends never[],
+  Result,
+>(
+  target: (this: This, ...args: Args) => Promise<Result>,
```

Comment:
Making the result a `Promise<Result>` I make sure that this is only used for async functions